src/Histogram.js

```diff
@@ -159,6 +161,61 @@ export default class Histogram {
     return this.lutGenerator_minMax(b, e);
   }
 
+  /**
+   * Generate a lookup table with a different color per intensity value
+   * @return {Lut}
+   */
+  lutGenerator_labelColors() {
+    const lut = new Uint8Array(256 * 4).fill(0);
+    const controlPoints = [];
+    controlPoints.push({ x: 0, opacity: 0, color: [0, 0, 0] });
+    let lastr = 0;
+    let lastg = 0;
+    let lastb = 0;
+    let lasta = 0;
+    let r = 0;
+    let g = 0;
+    let b = 0;
+    let a = 0;
+
+    // assumes exactly one bin per intensity value?
+    // skip zero!!!
+    for (let i = 1; i < this.bins.length; ++i) {
+      if (this.bins[i] > 0) {
+        const rgb = getColorByChannelIndex(i);
+
+        lut[i * 4 + 0] = rgb[0];
+        lut[i * 4 + 1] = rgb[1];
+        lut[i * 4 + 2] = rgb[2];
+        lut[i * 4 + 3] = 255;
+
+        r = rgb[0];
+        g = rgb[1];
+        b = rgb[2];
+        a = 1;
+      } else {
+        // add a zero control point?
+        r = 0;
+        g = 0;
+        b = 0;
+        a = 0;
+      }
+      // if current control point is same as last one don't add it
+      if (r !== lastr || g !== lastg || b !== lastb || a !== lasta) {
```

Comment:
Does this mean that the values in `this.bin` are sorted? i.e. we don't need to compare the rgba values with any other past values than the last set of values?

Reply:
if there are a bunch of empty bins in between bins with data, we don't want to keep inserting the same control point over and over.  Basically we only want to add a control point if its information would be DIFFERENT than the previous one.  To keep the number of control points as low as possible.  The control points get connected linearly.  Which is now making me think that this algorithm is going to have some errors. I really need to make a unit test for this control point + LUT stuff.